### ISSUE TITLE
Items used by PC can be sorted away by NPC

### ISSUE THREAD

**Issue by Funguss** [CleverRaven/Cataclysm-DDA#38459]
This is most obvious when butchering but I suspect it would apply in other situations, e.g. crafting or construction, or if the roles are reversed (PC sorting and NPC butchering). If an NPC is set to sort items and the player is butchering a corpse that is in an unsorted zone, the corpse can be moved. The player continues butchering until the task is complete, at which point an error is thrown.

# Steps To Reproduce

1. Place corpse in unsorted loot zone with a corpse loot zone nearby.
2. Tell NPC to sort items (ensure they won't be able to move the corpse before step 3)
3. Butcher corpse.

# Expected behavior

This was an unusual situation and wouldn't likely come up often, but I expect with future changes it could become more frequent and annoying.

The best fix I can come up with is tagging items, tools and furniture that are being used in an activity (woodcutting, construction, crafting, etc.) as being "in use". Any attempt to move these items will need a corresponding flag which allows this. Items tagged as "in use" should update the activity with a warning message when the items are lost or destroyed, prompting the player to cancel or continue using a different object, where appropriate.

# Versions and configuration

# Additional context

- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on unknown Windows version
- Game Version: 0.D-12504-g9015618 [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Aftershock [aftershock],
    C.R.I.T Expansion Mod [crt_expansion],
    Magiclysm [magiclysm],
    Makeshift Items Mod [makeshift],
    More Survival Tools [More_Survival_Tools],
    Mythological Replicas [nw_pack],
    Hydroponics [hydroponics],
    Mutant NPCs [mutant_npcs],
    No Rail Stations [No_Rail_Stations],
    Folding Parts pack [deoxymod],
    Vehicle Additions Pack [blazemod],
    Tanks and Other Vehicles [Tanks],
    Roadheader and other mining vehicles [Heavy miners],
    No Fungal Monsters [No_Fungi],
    Beta National Guard Camp [national_guard_camp]
]

**Comment by davidpwbrown** [comment]
Yeah, I can add the same kinda scan I do for NPC vehicle deconstruction, to sorting, so that they scan all other NPCs and player activity targets in the area, to ensure its not being worked on before they sort it. 

Ill do that today if possible.
